packages/notebook/src/widget.ts

```diff
@@ -892,6 +900,44 @@ export class StaticNotebook extends WindowedList {
     }
   }
 
+  private async _updateForDeferMode(
+    cell: Cell<ICellModel>,
+    cellIdx: number
+  ): Promise<void> {
+    cell.dataset.windowedListIndex = `${cellIdx}`;
+    this.layout.insertWidget(cellIdx, cell);
+    await cell.ready;
+  }
+
+  private _renderCSSAndJSOutputs(
+    cell: Cell<ICellModel>,
+    cellIdx: number
+  ): void {
+    // Only render cell with text/html outputs containing scripts or/and styles
+    // Note:
+    //   We don't need to render JavaScript mimetype outputs because they get
+    //   directly evaluate without adding DOM elements (see @jupyterlab/javascript-extension)
+    if (cell instanceof CodeCell) {
+      for (
+        let outputIdx = 0;
+        outputIdx < cell.model.outputs.length;
+        outputIdx++
+      ) {
+        const output = cell.model.outputs.get(outputIdx);
+        const html =
+          (output.data[htmlRendererFactory.mimeTypes[0]] as string) ?? '';
```

Comment:
What happens if another `mimeType` is added to `htmlRendererFactory`? Should we either;
- hard-code `html/text` to have agreement with what is written in user-facing FAQ, or
- iterate over all `mimeTypes`

Reply:
I would rather limit to `html/text`.